handsontable/src/plugins/dialog/dialog.js

```diff
@@ -0,0 +1,471 @@
+import { BasePlugin } from '../base';
+import { Hooks } from '../../core/hooks';
+import { removeClass, addClass, hasClass, fastInnerHTML, setAttribute } from '../../helpers/dom/element';
+import { warn } from '../../helpers/console';
+import { A11Y_DIALOG, A11Y_MODAL } from '../../helpers/a11y';
+
+export const PLUGIN_KEY = 'dialog';
+export const PLUGIN_PRIORITY = 340;
+const DIALOG_CLASS_NAME = 'ht-dialog';
+const SHORTCUTS_GROUP = PLUGIN_KEY;
+const SHORTCUTS_CONTEXT_NAME = `plugin:${PLUGIN_KEY}`;
+
+// Register dialog hooks
+Hooks.getSingleton().register('beforeDialogShow');
```

Comment:
There is a missing doc for those four hooks. That file should be updated https://github.com/handsontable/handsontable/blob/feature/dev-issue-2707/handsontable/src/core/hooks/constants.js.